### PR TITLE
feat: add gestures utility

### PR DESCRIPTION
**Pull request by mlijanto**
# Description

Initial merge to add PointerGestureHandler, a utility library for device-agnostic gesture detection.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/utilities/fast-gestures/src/utilities/get-rotation-angle.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export function getRotationAngle(x1: number, y1: number, x2: number, y2: number): number {
+  const angle: number = (Math.atan2(y1 - y2, x1 - x2) * 180) / Math.PI + 180;
```
Comment (nicholasrice) [inline packages/utilities/fast-gestures/src/utilities/get-rotation-angle.ts:2]: Minor nit; I don't think you need to allocate a variable here

File: packages/utilities/fast-gestures/src/utilities/get-coordinate-type.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    case PointerCoordinateType.Page:
+      return { x: e.pageX, y: e.pageY };
+
+    case PointerCoordinateType.Offset:
+      return { x: e.offsetX, y: e.offsetY };
+
+    case PointerCoordinateType.Screen:
+      return { x: e.screenX, y: e.screenY };
+
```
Comment (nicholasrice) [inline packages/utilities/fast-gestures/src/utilities/get-coordinate-type.ts:17]: `PointerCoordinateType.Page` and `default` can be combined

File: packages/utilities/fast-gestures/docs/api-report.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+//
+// @public (undocumented)
+export type IPointerGestureConfig = Partial<Record<PointerGestureFeature, boolean>>;
+
+// Warning: (ae-forgotten-export) The symbol "IPointerGestureInfo" needs to be exported by the entry point index.d.ts
+//
+// @public (undocumented)
+export type PointerGestureCallback = (pointerGestureInfo: IPointerGestureInfo) => void | null;
+
+// Warning: (ae-forgotten-export) The symbol "Disposable" needs to be exported by the entry point index.d.ts
```
Comment (nicholasrice) [inline packages/utilities/fast-gestures/docs/api-report.md:27]: Can you make sure these ae-forgotten-export subjects are exported?

**Comment by nicholasrice** [comment]
Awesome, thanks @mlijanto! This looks great. I'm noticing a break in the build when looking for the entry `/src/index.ts` file, do you know what that is about? 

https://github.com/microsoft/fast/pull/4441/checks?check_run_id=2079900033

File: packages/utilities/fast-gestures/src/interfaces/pointer-gesture-info.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { PointerTarget } from "../pointer-target";
+import { IPointerInfo } from "./pointer-info";
+
+export interface IPointerGestureInfo {
```
Comment (chrisdholt) [inline packages/utilities/fast-gestures/src/interfaces/pointer-gesture-info.ts:4]: @mlijanto can we drop "I" as a prefix to the interfaces across the PR?

File: packages/utilities/fast-gestures/docs/api-report.md (hunk 3 above)
Comment (mlijanto) [reply packages/utilities/fast-gestures/docs/api-report.md]: Sure. I've addressed all the `ae-forgotten-export` warnings. Thanks!

**Comment by mlijanto** [comment]
> Awesome, thanks @mlijanto! This looks great. I'm noticing a break in the build when looking for the entry `/src/index.ts` file, do you know what that is about?
> 
> https://github.com/microsoft/fast/pull/4441/checks?check_run_id=2079900033

Thanks for the review, @nicholasrice! Looks like the build break is caused by the empty `@microsoft/fast-physics` package. It can't find an entry point for the package. Is there a way we can skip building that package for now?

File: packages/utilities/fast-gestures/src/interfaces/pointer-gesture-info.ts (hunk 4 above)
Comment (mlijanto) [reply packages/utilities/fast-gestures/src/interfaces/pointer-gesture-info.ts]: @chrisdholt I've removed the "I" prefix from most of the interfaces, but there are some cases where removing the prefix would create a conflicting name between the interface and the class that implements it. Could I leave the "I" prefix in these cases, or do you have any guidance or pattern I should follow?

Comment (chrisdholt) [reply packages/utilities/fast-gestures/src/interfaces/pointer-gesture-info.ts]: Can you point to those scenarios? I see similar names but I don't see any with conflicts - I could be missing it.

Comment (mlijanto) [reply packages/utilities/fast-gestures/src/interfaces/pointer-gesture-info.ts]: There's a few of them, such as `IPoint` and `Point`, `IEvent` and `Event`. But this is no longer an issue. Rob and Nick gave a guidance to add the prefix "Default" to the implementations so I could remove the "I" prefix from all the interfaces. Thanks!

**Comment by awentzel** [comment]
It looks like this is ready to be completed.  Pinging for review.

**Review by nicholasrice — APPROVED** [review]

**Comment by mlijanto** [comment]
> It looks like this is ready to be completed. Pinging for review.

There's still a couple of more works need to be done after our last discussion with Rob and Nick. I'll need to replace the dependencies on codes taken from the VSCode project with our own and write some more unit tests.

**Comment by EisenbergEffect** [comment]
@mlijanto Where are we at on this? Is this ready for a review/merge/publish? Or does it need more work?

**Pull request merged**